File: CLASS ACTIVITY/10.7.2026/s3_hospital_admissions.py
import matplotlib.pyplot as plt
import seaborn as sns
import pandas as pd

sns.set_style("whitegrid")

depts = ['Cardiology', 'Orthopedics', 'Pediatrics', 'Neurology', 'ENT']
patients = [120, 95, 150, 60, 80]
cost = [45000, 32000, 18000, 52000, 15000]
df3 = pd.DataFrame({'Department': depts, 'Patients': patients, 'Cost': cost})

plt.figure(figsize=(7, 5))
ax = sns.barplot(data=df3, x='Department', y='Patients', hue='Department', palette='Set2', legend=False)
for i, c in enumerate(cost):
    ax.text(i, patients[i] + 2, f"₹{c}", ha='center', fontsize=9)
plt.title("Patient Admissions by Department (label=Avg Treatment Cost)")
plt.tight_layout()
plt.show()

File: CLASS ACTIVITY/10.7.2026/RP/s3_hospital_admissions.R
library(ggplot2)

depts <- c('Cardiology','Orthopedics','Pediatrics','Neurology','ENT')
patients <- c(120,95,150,60,80)
cost <- c(45000,32000,18000,52000,15000)
df3 <- data.frame(Department=depts, Patients=patients, Cost=cost)

ggplot(df3, aes(x=Department, y=Patients, fill=Department)) +
  geom_col() +
  geom_text(aes(label=paste0("Rs.", Cost)), vjust=-0.4, size=3.3) +
  labs(title="Patient Admissions by Department (label=Avg Treatment Cost)",
       y="Number of Patients") +
  theme_minimal() +
  theme(legend.position="none")
